src/internal/m365/collection/site/restore.go

```diff
@@ -54,25 +62,100 @@ func restoreListItem(
 		return dii, clues.WrapWC(ctx, err, "generating list from stored bytes")
 	}
 
-	newName := formatListsRestoreDestination(destName, itemID, storedList)
+	var (
+		collisionKey = api.ListCollisionKey(storedList)
+		collisionID  string
+		restoredList models.Listable
+		newName      = formatListsRestoreDestination(destName, itemID, storedList)
+	)
 
-	// Restore to List base to M365 back store
-	restoredList, err := rh.PostList(ctx, newName, storedList, errs)
-	if err != nil {
-		return dii, graph.Wrap(ctx, err, "restoring list")
+	if id, ok := collisionKeyToItemID[collisionKey]; ok {
+		log := logger.Ctx(ctx).With("collision_key", clues.Hide(collisionKey))
+		log.Debug("item collision")
+
+		if collisionPolicy == control.Skip {
+			ctr.Inc(count.CollisionSkip)
+			log.Debug("skipping item with collision")
+
+			return dii, core.ErrAlreadyExists
+		}
+
+		collisionID = id
 	}
 
+	if collisionPolicy != control.Replace {
+		restoredList, err = rh.PostList(ctx, newName, storedList, errs)
+		if err != nil {
+			return dii, clues.WrapWC(ctx, err, "restoring list")
+		}
+	} else {
+		restoredList, err = handleListReplace(
+			ctx,
+			collisionID,
+			storedList,
+			rh,
+			ctr,
+			errs)
+		if err != nil {
+			return dii, err
+		}
+	}
+
+	// Restore to List base to M365 back store
+
 	dii.SharePoint = api.ListToSPInfo(restoredList)
 
 	return dii, nil
 }
 
+func handleListReplace(
+	ctx context.Context,
+	collisionID string,
+	storedList models.Listable,
+	rh restoreHandler,
+	ctr *count.Bus,
+	errs *fault.Bus,
+) (models.Listable, error) {
+	collidedList, _, err := rh.GetList(ctx, collisionID)
+	if err != nil {
+		return nil, clues.WrapWC(ctx, err, "fetching collided list")
+	}
+
+	err = rh.DeleteList(ctx, collisionID)
+	if err != nil {
+		return nil, clues.WrapWC(ctx, err, "deleting collided list")
+	}
+
+	restoredList, err := rh.PostList(
+		ctx,
+		ptr.Val(storedList.GetDisplayName()),
+		storedList,
+		errs)
+	if err == nil {
+		ctr.Inc(count.CollisionReplace)
+		return restoredList, nil
+	}
+
+	_, collidedListErr := rh.PostList(
```

Comment:
I'm confused by what's going on with the behavior here.  As a general statement, extra comments are always helpful for other maintainers so that we can understand what's going on in a process.  This is one of those places where a little explanation would go a long way.

So, what I figure is happening is that the func does this:

1. look up the current list
2. delete the current list
3. restore the backed-up list
4. if no error occurs when restoring, good, we exit.
5. if an error occurs when restoring, try to restore the list that we just deleted.

I think I have that right.  If so: that works, but it's not my favorite pattern, because it contains a lot more moving parts and points of failure.  I think you could slim it down to the following for a simpler flow and fewer points of failure.  At least, I think this is possible, assuming the newly created list has a different ID than the previous one.

1. Restore the backed-up list.  If an error occurs, return.
2. Delete the current list.  If an error occurs, return.

And if that works, we might not need this extra func at all.  Up on lines ~85-105 you could do this instead:
```go
 restoredList, err = rh.PostList(ctx, newName, storedList, errs)
 if err != nil {
 	return dii, clues.WrapWC(ctx, err, "restoring list")
 }

if collisionPolicy != control.Replace {
  rh.DeletetList(...)
}
```

Reply:
@ryanfkeepers 
you got it right. I would have gone for the pattern you mentioned as well, but in `replace` collision, we need to create backed-ip list with same name as current/collided list. But site does not allow to create two lists with same name.

Reply:
Thanks for confirming!  I grok the need to work around it.  I pitched a third workflow option in the other comment; let me know what you think.